feat: Add different gap sizes for if a workspace has only one window

Miitto opened the pull request:
Added a new config option under `gaps` called `single_screen_gap` (probably want a different name) which sets the outer gap sizes for any workspace with only one tiling window in it.

Also fixed a possible bug where reading the config would fail if the home directory could not be found, even if a custom config path was passed that pointed elsewhere.

lars-berger commented:
Ay thanks for the PR’s ❤️ Rushing a bit currently to get a new release out with the latest batch of fixes + features. I can do some proper testing and review these after - shouldn’t be more than a couple of days

Miitto commented:
Annoying force push there, it was that or a merge commit.  Got some commits out of order with a rebase earlier I think.
Can't find anything overwritten, and all the commits made are there.

Any tips on how to avoid in future? I think something to do with cherry picking might help.

lars-berger commented on packages/wm-common/src/parsed_config.rs:
```diff
@@ -220,20 +235,47 @@ pub struct WindowEffectsConfig {
 pub struct WindowEffectConfig {
   /// Config for optionally applying a colored border.
-  pub border: BorderEffectConfig,
+  pub border: SmartBorderEffectConfig,
```
i'm fully on board with the gaps feature, but there's a few things to figure out re. the smart borders. is it mainly for the sake of if someone is running a single window outer gap of 0px? trying to get a better understanding of the use case.

if the purpose is primarily to remove border/corner effects if the window takes up the full monitor, maybe it'd make more sense to have an option for disabling window effects if a window is fullscreen or is a single window with an outer gap of 0px? either way, imo the best would be to figure out the exact use case and then potentially pr a solution for that separately

Miitto commented:
Removed border effects etc, and switched to using the config getters

`PositionGetters::to_rect` for `Workspace` is still done without a getter since it doesn't have access to the full `UserConfig`

lars-berger commented:
> Removed border effects etc, and switched to using the config getters
> 
> `PositionGetters::to_rect` for `Workspace` is still done without a getter since it doesn't have access to the full `UserConfig`

Really well done 🎉 I happen to be doing some final prep before doing a new release, so we might be able to squeeze this in.

Miitto commented on packages/wm/src/user_config.rs:
```diff
@@ -38,12 +38,14 @@ impl UserConfig {
     let config_path = config_path
       .or_else(|| env::var("GLAZEWM_CONFIG_PATH").ok().map(PathBuf::from))
-      .unwrap_or(default_config_path);
+      .unwrap_or(default_config_path?);
```
I can't remember exactly what I was doing anymore sadly, bit I did manage to get the `home_dir` call to fail. Although at that point the system has bigger problems.
